experimental/regression_suite/ireers_tools/artifacts.py

```diff
@@ -228,7 +213,8 @@ def download_azure_artifact(self: "FetchedArtifact"):
 
     @staticmethod
     def _callback(self: "FetchedArtifact"):
-        self.download_azure_artifact()
+        if "blob.core.windows.net" in self.url:
+            self.download_azure_artifact()
```

Comment:
else raise an exception, suggested message:

```suggestion
        if "blob.core.windows.net" in self.url:
            self.download_azure_artifact()
        else:
            raise NotImplementedError(f"Unsupported fetched artifact URL schema for '{self.url}'")
            
```